simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -791,75 +791,69 @@ auto HdMapUtils::getLaneChangeableLaneletId(
+      total_distance = total_distance + getLaneletLength(straight_lanelet_ids[0]);
+      previous_lanelets_ids.push_back(straight_lanelet_ids[0]);
+    } else if (auto non_straight_lanelet_ids = getPreviousLaneletIds(reference_lanelet_id);
+               not non_straight_lanelet_ids.empty()) {
+      total_distance = total_distance + getLaneletLength(non_straight_lanelet_ids[0]);
+      previous_lanelets_ids.push_back(non_straight_lanelet_ids[0]);
     } else {
-      auto else_ids = getPreviousLaneletIds(lanelet_id);
-      if (else_ids.size() != 0) {
-        total_distance = total_distance + getLaneletLength(else_ids[0]);
-        ret.push_back(else_ids[0]);
-        continue;
-      } else {
-        break;
-      }
+      break;
     }
   }
-  return ret;
+  return previous_lanelets_ids;
 }
 
 auto HdMapUtils::isInRoute(const lanelet::Id lanelet_id, const lanelet::Ids & route) const -> bool
 {
-  return std::find_if(route.begin(), route.end(), [lanelet_id](const auto id) {
-           return lanelet_id == id;
-         }) != route.end();
+  return std::find(route.cbegin(), route.cend(), lanelet_id) != route.cend();
 }
 
 auto HdMapUtils::getFollowingLanelets(
-  const lanelet::Id lanelet_id, const lanelet::Ids & candidate_lanelet_ids, const double distance,
-  const bool include_self) const -> lanelet::Ids
-{
-  if (candidate_lanelet_ids.empty()) {
-    return {};
-  }
-  lanelet::Ids ids;
-  double total_distance = 0.0;
-  bool found = false;
-  for (const auto id : candidate_lanelet_ids) {
-    if (found) {
-      ids.emplace_back(id);
-      total_distance = total_distance + getLaneletLength(id);
-      if (total_distance > distance) {
-        return ids;
+  const lanelet::Id current_lanelet_id, const lanelet::Ids & route, const double horizont,
+  const bool include_current_lanelet_id) const -> lanelet::Ids
+{
+  lanelet::Ids following_lanelets_ids;
+  if (not route.empty()) {
+    double total_distance = 0.0;
+    for (const auto & candidate_lanelet_id : route) {
+      if (not following_lanelets_ids.empty()) {
+        if (const auto next_ids = getNextLaneletIds(following_lanelets_ids.back());
+            std::find(next_ids.cbegin(), next_ids.cend(), candidate_lanelet_id) ==
+            next_ids.cend()) {
+          THROW_SEMANTIC_ERROR("lanelet id does not match");
+        }
+        following_lanelets_ids.push_back(candidate_lanelet_id);
+        total_distance += getLaneletLength(candidate_lanelet_id);
+        if (total_distance > horizont) {
+          break;
+        }
+      } else if (candidate_lanelet_id == current_lanelet_id) {
+        if (include_current_lanelet_id) {
+          following_lanelets_ids.push_back(candidate_lanelet_id);
+        }
       }
     }
-    if (id == lanelet_id) {
-      found = true;
-      if (include_self) {
-        ids.push_back(id);
-      }
+    if (following_lanelets_ids.empty()) {
+      THROW_SEMANTIC_ERROR("lanelet id does not match");
+    } else if (total_distance < horizont) {
```

Comment:
was `total_distance > distance` so here it should be `total_distance <= horizon`

Reply:
done